gnome-initial-setup/gis-driver.c

```diff
@@ -80,12 +81,40 @@ struct _GisDriverPrivate {
   gchar *personality;
   gchar *lang_id;
 
+  gboolean is_live_session;
+
   GisDriverMode mode;
 };
 typedef struct _GisDriverPrivate GisDriverPrivate;
 
 G_DEFINE_TYPE_WITH_PRIVATE(GisDriver, gis_driver, GTK_TYPE_APPLICATION)
 
+static gboolean
+running_live_session (void)
+{
+  gboolean has_live_boot_param;
+  GError *error;
+  gchar *contents;
```

Comment:
If we are keen on autoptrs these days, this could be `g_autofree gchar *contents = NULL`.